src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftPermanentStorage.java

```diff
@@ -207,18 +222,91 @@ public boolean delete(SapsImage task) {
                 ? properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_DEBUG_TASKS_DIR)
                 : properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_TASKS_DIR);
 
-        String prefix = swiftExports + File.separator + taskId;
+        String dir = swiftExports + File.separator + taskId;
 
-        List<String> fileNames = swiftAPIClient.listFilesWithPrefix(containerName, prefix);
+        try {
+            List<String> fileNames = swiftAPIClient.listFiles(containerName, dir);
+            LOGGER.info("Files List: " + fileNames);
 
-        LOGGER.info("Files List: " + fileNames);
-
-        for (String file : fileNames) {
-            LOGGER.debug("Trying to delete file " + file + " from " + containerName);
-            swiftAPIClient.deleteFile(containerName, file);
+            for (String file : fileNames) {
+                LOGGER.debug("Trying to delete file " + file + " from " + containerName);
+                swiftAPIClient.deleteFile(containerName, file);
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error while list files of path [" + dir + "] from Object Storage", e);
+            return false;
         }
 
         return true;
     }
 
+    @Override
+    public List<AccessLink> generateAccessLinks(String taskId) {
+        List<AccessLink> filesLinks = new ArrayList<>();
+        List<String> files = new ArrayList<>();
+
+        String tasksDir = properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_TASKS_DIR);
+        String[] dirs = {INPUTDOWNLOADING_DIR, PREPROCESSING_DIR, PROCESSING_DIR};
+
+        for (String dir : dirs) {
+            String dirPath = String.format(SWIFT_TASK_STAGE_DIR_PATTERN, tasksDir, taskId, dir);
+            try {
+                files.addAll(this.swiftAPIClient.listFiles(this.containerName, dirPath));
+            } catch (IOException e) {
+                LOGGER.error("Error while list files of path [" + dir + "] from Object Storage", e);
+            }
+        }
+
```

Comment:
this method could be divided into two smaller ones: `listFiles` and `generateLinks`

Reply:
private ones

Reply:
Done.